exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/xsort/ExternalSortBatch.java

```diff
@@ -550,7 +565,15 @@ public BatchGroup mergeAndSpill(LinkedList<BatchGroup> batchGroups) throws Schem
     c1.buildSchema(BatchSchema.SelectionVectorMode.NONE);
     c1.setRecordCount(count);
 
-    String outputFile = Joiner.on("/").join(dirs.next(), fileName, spillCount++);
+    String spillDir = dirs.next();
+    Path currSpillPath = new Path(Joiner.on("/").join(spillDir, fileName));
+    currSpillDirs.add(currSpillPath);
+    String outputFile = Joiner.on("/").join(currSpillPath, spillCount++);
+    try {
+        fs.deleteOnExit(currSpillPath);
+    } catch (IOException e) {
+        throw new RuntimeException(e);
```

Comment:
I have some concerns about throwing an exception here:

First, does it make sense to fail the query when `deleteOnExit()` fails ? shouldn't we just log a warning. After all, if all goes well, this folder will get deleted in the close method.

Second, if this exception throws we'll be leaking memory because we'll skip clearing batchGroupList and hyperBatch